File: msgraph-mail/go/utilities/models/message_rule_actions.go
package models

import (
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91 "github.com/microsoft/kiota-abstractions-go/serialization"
)

// MessageRuleActions 
type MessageRuleActions struct {
    // Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additionalData map[string]interface{}
    // A list of categories to be assigned to a message.
    assignCategories []string
    // The ID of a folder that a message is to be copied to.
    copyToFolder *string
    // Indicates whether a message should be moved to the Deleted Items folder.
    delete *bool
    // The email addresses of the recipients to which a message should be forwarded as an attachment.
    forwardAsAttachmentTo []Recipientable
    // The email addresses of the recipients to which a message should be forwarded.
    forwardTo []Recipientable
    // Indicates whether a message should be marked as read.
    markAsRead *bool
    // The markImportance property
    markImportance *Importance
    // The ID of the folder that a message will be moved to.
    moveToFolder *string
    // Indicates whether a message should be permanently deleted and not saved to the Deleted Items folder.
    permanentDelete *bool
    // The email address to which a message should be redirected.
    redirectTo []Recipientable
    // Indicates whether subsequent rules should be evaluated.
    stopProcessingRules *bool
}
// NewMessageRuleActions instantiates a new messageRuleActions and sets the default values.
func NewMessageRuleActions()(*MessageRuleActions) {
    m := &MessageRuleActions{
    }
    m.SetAdditionalData(make(map[string]interface{}));
    return m
}
// CreateMessageRuleActionsFromDiscriminatorValue creates a new instance of the appropriate class based on discriminator value
func CreateMessageRuleActionsFromDiscriminatorValue(parseNode i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, error) {
    return NewMessageRuleActions(), nil
}
// GetAdditionalData gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
func (m *MessageRuleActions) GetAdditionalData()(map[string]interface{}) {
    return m.additionalData
}
// GetAssignCategories gets the assignCategories property value. A list of categories to be assigned to a message.
func (m *MessageRuleActions) GetAssignCategories()([]string) {
    return m.assignCategories
}
// GetCopyToFolder gets the copyToFolder property value. The ID of a folder that a message is to be copied to.
func (m *MessageRuleActions) GetCopyToFolder()(*string) {
    return m.copyToFolder
}
// GetDelete gets the delete property value. Indicates whether a message should be moved to the Deleted Items folder.
func (m *MessageRuleActions) GetDelete()(*bool) {
    return m.delete
}
// GetFieldDeserializers the deserialization information for the current model
func (m *MessageRuleActions) GetFieldDeserializers()(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error)) {
    res := make(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error))
    res["assignCategories"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetCollectionOfReferencedPrimitiveValue(n.GetCollectionOfPrimitiveValues , string , m.SetAssignCategories)
    }
    res["copyToFolder"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetValue(n.GetStringValue , m.SetCopyToFolder)
    }
    res["delete"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetValue(n.GetBoolValue , m.SetDelete)
    }
    res["forwardAsAttachmentTo"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetCollectionValue(n.GetCollectionOfObjectValues , CreateRecipientFromDiscriminatorValue , m.SetForwardAsAttachmentTo)
    }
    res["forwardTo"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetCollectionValue(n.GetCollectionOfObjectValues , CreateRecipientFromDiscriminatorValue , m.SetForwardTo)
    }
    res["markAsRead"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetValue(n.GetBoolValue , m.SetMarkAsRead)
    }
    res["markImportance"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetReferencedEnumValue(n.GetEnumValue , ParseImportance , m.SetMarkImportance)
    }
    res["moveToFolder"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetValue(n.GetStringValue , m.SetMoveToFolder)
    }
    res["permanentDelete"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetValue(n.GetBoolValue , m.SetPermanentDelete)
    }
    res["redirectTo"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetCollectionValue(n.GetCollectionOfObjectValues , CreateRecipientFromDiscriminatorValue , m.SetRedirectTo)
    }
    res["stopProcessingRules"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
        return i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetValue(n.GetBoolValue , m.SetStopProcessingRules)
    }
    return res
}
// GetForwardAsAttachmentTo gets the forwardAsAttachmentTo property value. The email addresses of the recipients to which a message should be forwarded as an attachment.
func (m *MessageRuleActions) GetForwardAsAttachmentTo()([]Recipientable) {
    return m.forwardAsAttachmentTo
}
// GetForwardTo gets the forwardTo property value. The email addresses of the recipients to which a message should be forwarded.
func (m *MessageRuleActions) GetForwardTo()([]Recipientable) {
    return m.forwardTo
}
// GetMarkAsRead gets the markAsRead property value. Indicates whether a message should be marked as read.
func (m *MessageRuleActions) GetMarkAsRead()(*bool) {
    return m.markAsRead
}
// GetMarkImportance gets the markImportance property value. The markImportance property
func (m *MessageRuleActions) GetMarkImportance()(*Importance) {
    return m.markImportance
}
// GetMoveToFolder gets the moveToFolder property value. The ID of the folder that a message will be moved to.
func (m *MessageRuleActions) GetMoveToFolder()(*string) {
    return m.moveToFolder
}
// GetPermanentDelete gets the permanentDelete property value. Indicates whether a message should be permanently deleted and not saved to the Deleted Items folder.
func (m *MessageRuleActions) GetPermanentDelete()(*bool) {
    return m.permanentDelete
}
// GetRedirectTo gets the redirectTo property value. The email address to which a message should be redirected.
func (m *MessageRuleActions) GetRedirectTo()([]Recipientable) {
    return m.redirectTo
}
// GetStopProcessingRules gets the stopProcessingRules property value. Indicates whether subsequent rules should be evaluated.
func (m *MessageRuleActions) GetStopProcessingRules()(*bool) {
    return m.stopProcessingRules
}
// Serialize serializes information the current object
func (m *MessageRuleActions) Serialize(writer i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.SerializationWriter)(error) {
    if m.GetAssignCategories() != nil {
        err := writer.WriteCollectionOfStringValues("assignCategories", m.GetAssignCategories())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteStringValue("copyToFolder", m.GetCopyToFolder())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteBoolValue("delete", m.GetDelete())
        if err != nil {
            return err
        }
    }
    if m.GetForwardAsAttachmentTo() != nil {
        cast := make([]i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, len(m.GetForwardAsAttachmentTo()))
        for i, v := range m.GetForwardAsAttachmentTo() {
            cast[i] = v.(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable)
        }
        err := writer.WriteCollectionOfObjectValues("forwardAsAttachmentTo", cast)
        if err != nil {
            return err
        }
    }
    if m.GetForwardTo() != nil {
        cast := make([]i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, len(m.GetForwardTo()))
        for i, v := range m.GetForwardTo() {
            cast[i] = v.(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable)
        }
        err := writer.WriteCollectionOfObjectValues("forwardTo", cast)
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteBoolValue("markAsRead", m.GetMarkAsRead())
        if err != nil {
            return err
        }
    }
    if m.GetMarkImportance() != nil {
        cast := (*m.GetMarkImportance()).String()
        err := writer.WriteStringValue("markImportance", &cast)
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteStringValue("moveToFolder", m.GetMoveToFolder())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteBoolValue("permanentDelete", m.GetPermanentDelete())
        if err != nil {
            return err
        }
    }
    if m.GetRedirectTo() != nil {
        cast := make([]i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, len(m.GetRedirectTo()))
        for i, v := range m.GetRedirectTo() {
            cast[i] = v.(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable)
        }
        err := writer.WriteCollectionOfObjectValues("redirectTo", cast)
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteBoolValue("stopProcessingRules", m.GetStopProcessingRules())
        if err != nil {
            return err
        }
    }
    {
        err := writer.WriteAdditionalData(m.GetAdditionalData())
        if err != nil {
            return err
        }
    }
    return nil
}
// SetAdditionalData sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
func (m *MessageRuleActions) SetAdditionalData(value map[string]interface{})() {
    m.additionalData = value
}
// SetAssignCategories sets the assignCategories property value. A list of categories to be assigned to a message.
func (m *MessageRuleActions) SetAssignCategories(value []string)() {
    m.assignCategories = value
}
// SetCopyToFolder sets the copyToFolder property value. The ID of a folder that a message is to be copied to.
func (m *MessageRuleActions) SetCopyToFolder(value *string)() {
    m.copyToFolder = value
}
// SetDelete sets the delete property value. Indicates whether a message should be moved to the Deleted Items folder.
func (m *MessageRuleActions) SetDelete(value *bool)() {
    m.delete = value
}
// SetForwardAsAttachmentTo sets the forwardAsAttachmentTo property value. The email addresses of the recipients to which a message should be forwarded as an attachment.
func (m *MessageRuleActions) SetForwardAsAttachmentTo(value []Recipientable)() {
    m.forwardAsAttachmentTo = value
}
// SetForwardTo sets the forwardTo property value. The email addresses of the recipients to which a message should be forwarded.
func (m *MessageRuleActions) SetForwardTo(value []Recipientable)() {
    m.forwardTo = value
}
// SetMarkAsRead sets the markAsRead property value. Indicates whether a message should be marked as read.
func (m *MessageRuleActions) SetMarkAsRead(value *bool)() {
    m.markAsRead = value
}
// SetMarkImportance sets the markImportance property value. The markImportance property
func (m *MessageRuleActions) SetMarkImportance(value *Importance)() {
    m.markImportance = value
}
// SetMoveToFolder sets the moveToFolder property value. The ID of the folder that a message will be moved to.
func (m *MessageRuleActions) SetMoveToFolder(value *string)() {
    m.moveToFolder = value
}
// SetPermanentDelete sets the permanentDelete property value. Indicates whether a message should be permanently deleted and not saved to the Deleted Items folder.
func (m *MessageRuleActions) SetPermanentDelete(value *bool)() {
    m.permanentDelete = value
}
// SetRedirectTo sets the redirectTo property value. The email address to which a message should be redirected.
func (m *MessageRuleActions) SetRedirectTo(value []Recipientable)() {
    m.redirectTo = value
}
// SetStopProcessingRules sets the stopProcessingRules property value. Indicates whether subsequent rules should be evaluated.
func (m *MessageRuleActions) SetStopProcessingRules(value *bool)() {
    m.stopProcessingRules = value
}

File: msgraph-mail/ruby/graphrubyv4/models/microsoft/graph/message_rule_actions.rb
require './models'
require 'microsoft_kiota_abstractions'

module GraphC::Models
    class MessageRuleActions
        include IAdditionalDataHolder, MicrosoftKiotaAbstractions::Parsable
        ## 
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        @additional_data
        ## 
        # A list of categories to be assigned to a message.
        @assign_categories
        ## 
        # The ID of a folder that a message is to be copied to.
        @copy_to_folder
        ## 
        # Indicates whether a message should be moved to the Deleted Items folder.
        @delete
        ## 
        # The email addresses of the recipients to which a message should be forwarded as an attachment.
        @forward_as_attachment_to
        ## 
        # The email addresses of the recipients to which a message should be forwarded.
        @forward_to
        ## 
        # Indicates whether a message should be marked as read.
        @mark_as_read
        ## 
        # The markImportance property
        @mark_importance
        ## 
        # The ID of the folder that a message will be moved to.
        @move_to_folder
        ## 
        # Indicates whether a message should be permanently deleted and not saved to the Deleted Items folder.
        @permanent_delete
        ## 
        # The email address to which a message should be redirected.
        @redirect_to
        ## 
        # Indicates whether subsequent rules should be evaluated.
        @stop_processing_rules
        ## 
        ## Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        ## @return a i_dictionary
        ## 
        def  additional_data
            return @additional_data
        end
        ## 
        ## Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        ## @param value Value to set for the AdditionalData property.
        ## @return a void
        ## 
        def  additional_data=(additionalData)
            @additional_data = additionalData
        end
        ## 
        ## Gets the assignCategories property value. A list of categories to be assigned to a message.
        ## @return a string
        ## 
        def  assign_categories
            return @assign_categories
        end
        ## 
        ## Sets the assignCategories property value. A list of categories to be assigned to a message.
        ## @param value Value to set for the assignCategories property.
        ## @return a void
        ## 
        def  assign_categories=(assignCategories)
            @assign_categories = assignCategories
        end
        ## 
        ## Gets the copyToFolder property value. The ID of a folder that a message is to be copied to.
        ## @return a string
        ## 
        def  copy_to_folder
            return @copy_to_folder
        end
        ## 
        ## Sets the copyToFolder property value. The ID of a folder that a message is to be copied to.
        ## @param value Value to set for the copyToFolder property.
        ## @return a void
        ## 
        def  copy_to_folder=(copyToFolder)
            @copy_to_folder = copyToFolder
        end
        ## 
        ## Creates a new instance of the appropriate class based on discriminator value
        ## @param parseNode The parse node to use to read the discriminator value and create the object
        ## @return a message_rule_actions
        ## 
        def create_from_discriminator_value(parse_node) 
            return nil;
        end
        ## 
        ## Gets the delete property value. Indicates whether a message should be moved to the Deleted Items folder.
        ## @return a boolean
        ## 
        def  delete
            return @delete
        end
        ## 
        ## Sets the delete property value. Indicates whether a message should be moved to the Deleted Items folder.
        ## @param value Value to set for the delete property.
        ## @return a void
        ## 
        def  delete=(delete)
            @delete = delete
        end
        ## 
        ## Gets the forwardAsAttachmentTo property value. The email addresses of the recipients to which a message should be forwarded as an attachment.
        ## @return a recipient
        ## 
        def  forward_as_attachment_to
            return @forward_as_attachment_to
        end
        ## 
        ## Sets the forwardAsAttachmentTo property value. The email addresses of the recipients to which a message should be forwarded as an attachment.
        ## @param value Value to set for the forwardAsAttachmentTo property.
        ## @return a void
        ## 
        def  forward_as_attachment_to=(forwardAsAttachmentTo)
            @forward_as_attachment_to = forwardAsAttachmentTo
        end
        ## 
        ## Gets the forwardTo property value. The email addresses of the recipients to which a message should be forwarded.
        ## @return a recipient
        ## 
        def  forward_to
            return @forward_to
        end
        ## 
        ## Sets the forwardTo property value. The email addresses of the recipients to which a message should be forwarded.
        ## @param value Value to set for the forwardTo property.
        ## @return a void
        ## 
        def  forward_to=(forwardTo)
            @forward_to = forwardTo
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return {
                "assignCategories" => lambda {|o, n| o.assign_categories = n.get_collection_of_primitive_values(String) },
                "copyToFolder" => lambda {|o, n| o.copy_to_folder = n.get_string_value() },
                "delete" => lambda {|o, n| o.delete = n.get_boolean_value() },
                "forwardAsAttachmentTo" => lambda {|o, n| o.forward_as_attachment_to = n.get_collection_of_object_values(GraphC::Models::Recipient) },
                "forwardTo" => lambda {|o, n| o.forward_to = n.get_collection_of_object_values(GraphC::Models::Recipient) },
                "markAsRead" => lambda {|o, n| o.mark_as_read = n.get_boolean_value() },
                "markImportance" => lambda {|o, n| o.mark_importance = n.get_enum_value(GraphC::Models::Importance) },
                "moveToFolder" => lambda {|o, n| o.move_to_folder = n.get_string_value() },
                "permanentDelete" => lambda {|o, n| o.permanent_delete = n.get_boolean_value() },
                "redirectTo" => lambda {|o, n| o.redirect_to = n.get_collection_of_object_values(GraphC::Models::Recipient) },
                "stopProcessingRules" => lambda {|o, n| o.stop_processing_rules = n.get_boolean_value() },
            }
        end
        ## 
        ## Gets the markAsRead property value. Indicates whether a message should be marked as read.
        ## @return a boolean
        ## 
        def  mark_as_read
            return @mark_as_read
        end
        ## 
        ## Sets the markAsRead property value. Indicates whether a message should be marked as read.
        ## @param value Value to set for the markAsRead property.
        ## @return a void
        ## 
        def  mark_as_read=(markAsRead)
            @mark_as_read = markAsRead
        end
        ## 
        ## Gets the markImportance property value. The markImportance property
        ## @return a importance
        ## 
        def  mark_importance
            return @mark_importance
        end
        ## 
        ## Sets the markImportance property value. The markImportance property
        ## @param value Value to set for the markImportance property.
        ## @return a void
        ## 
        def  mark_importance=(markImportance)
            @mark_importance = markImportance
        end
        ## 
        ## Gets the moveToFolder property value. The ID of the folder that a message will be moved to.
        ## @return a string
        ## 
        def  move_to_folder
            return @move_to_folder
        end
        ## 
        ## Sets the moveToFolder property value. The ID of the folder that a message will be moved to.
        ## @param value Value to set for the moveToFolder property.
        ## @return a void
        ## 
        def  move_to_folder=(moveToFolder)
            @move_to_folder = moveToFolder
        end
        ## 
        ## Gets the permanentDelete property value. Indicates whether a message should be permanently deleted and not saved to the Deleted Items folder.
        ## @return a boolean
        ## 
        def  permanent_delete
            return @permanent_delete
        end
        ## 
        ## Sets the permanentDelete property value. Indicates whether a message should be permanently deleted and not saved to the Deleted Items folder.
        ## @param value Value to set for the permanentDelete property.
        ## @return a void
        ## 
        def  permanent_delete=(permanentDelete)
            @permanent_delete = permanentDelete
        end
        ## 
        ## Gets the redirectTo property value. The email address to which a message should be redirected.
        ## @return a recipient
        ## 
        def  redirect_to
            return @redirect_to
        end
        ## 
        ## Sets the redirectTo property value. The email address to which a message should be redirected.
        ## @param value Value to set for the redirectTo property.
        ## @return a void
        ## 
        def  redirect_to=(redirectTo)
            @redirect_to = redirectTo
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            writer.write_collection_of_primitive_values("assignCategories", @assign_categories)
            writer.write_string_value("copyToFolder", @copy_to_folder)
            writer.write_boolean_value("delete", @delete)
            writer.write_collection_of_object_values("forwardAsAttachmentTo", @forward_as_attachment_to)
            writer.write_collection_of_object_values("forwardTo", @forward_to)
            writer.write_boolean_value("markAsRead", @mark_as_read)
            writer.write_enum_value("markImportance", @mark_importance)
            writer.write_string_value("moveToFolder", @move_to_folder)
            writer.write_boolean_value("permanentDelete", @permanent_delete)
            writer.write_collection_of_object_values("redirectTo", @redirect_to)
            writer.write_boolean_value("stopProcessingRules", @stop_processing_rules)
            writer.write_additional_data(@additional_data)
        end
        ## 
        ## Gets the stopProcessingRules property value. Indicates whether subsequent rules should be evaluated.
        ## @return a boolean
        ## 
        def  stop_processing_rules
            return @stop_processing_rules
        end
        ## 
        ## Sets the stopProcessingRules property value. Indicates whether subsequent rules should be evaluated.
        ## @param value Value to set for the stopProcessingRules property.
        ## @return a void
        ## 
        def  stop_processing_rules=(stopProcessingRules)
            @stop_processing_rules = stopProcessingRules
        end
    end
end
